repository.py

```diff
@@ -844,10 +844,38 @@ def updateInstallProgress(m):
                 total = 0
                 updateInstallProgress(m)
         rv = p.wait()
+        gpg_error_message = None
+
+            if stderr.find(' in import_key_to_pubring') >= 0:
```

Comment:
Why this extra indentation?

Reply:
It's not really easy for me to do such forward-porting on `master` (especially testing), since we've not started to use that branch.  Would it be possible to target xs8 first, and then once that gets accepted start to work on a `master` version, to streamline the process a bit more?

Reply:
Sure, fine with me